packages/core/src/threatComposer/messageHandlers/addFileWatchMessageHandler.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Command, FileChangedMessage, MessageType, WebviewContext } from '../types'
+import vscode from 'vscode'
+import { fsCommon } from '../../srcShared/fs'
+
+/**
+ * Function to add a watcher on the file that was opened. The watcher will notify Threat Composer
+ * view when a change occurs to the file externally.
+ * If the change occurs due to
+ *    a user save action, file change is ignored.
+ *    an auto save that is persisted as AutoSave is turned on in VSCode settings, update file state.
+ *    an external file change,
+ *       if a local, unsaved change exist in Threat Composer, the file change is ignored.
+ *       if no unsaved local changes exist, the view is notified of the external file change.
+ * @param context: The Webview Context that contain the details of the file and the webview
+ */
+export function addFileWatchMessageHandler(context: WebviewContext) {
+    const filePath = context.defaultTemplatePath
+    const fileName = context.defaultTemplateName
+
+    context.disposables.push(
+        vscode.workspace.onDidChangeTextDocument(async e => {
```

Comment:
please consider [subclassing WatchedFiles](https://github.com/aws/aws-toolkit-vscode/blob/f5ed26f85a58afd3e248110ad5a1ee746fd057d8/packages/core/src/shared/fs/devfileRegistry.ts#L16) 

this allows us to have the same interface, logging, and troubleshooting steps for "watched files". if there are missing features from WatchedFiles, let's discuss